File: Funciones_Python/robot_connect.py
# udp_connection.py

import socket

# Variables globales del módulo
u = None
esp32_ip = None
esp32_port = None

def robot_connect():
    global u, esp32_ip, esp32_port

    if u is None:
        esp32_ip = "192.168.50.222"
        esp32_port = 12345

        try:
            u = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
            print('Conexión UDP establecida correctamente.')
        except Exception as e:
            print('ERROR: No se pudo conectar al robot. Verifica que esté encendido.')
            print(e)
    else:
        print('Ya existe una conexión UDP activa.')
    
    return u, esp32_ip, esp32_port  # Para usar el socket desde otro archivo

File: Funciones_Matlab/robot_connect.m
function robot_connect()
    % Establece la conexión con el ESP32 vía UDP y guarda los datos

    global u esp32_ip esp32_port

    if isempty(u)
        esp32_ip = "192.168.50.222";  % Estos 2 datos están así en el ESP32 del robot
        esp32_port = 12345;

        try
            u = udpport("datagram", "IPV4");
            disp('Conexión UDP establecida correctamente.');
        catch ME
            disp('ERROR: No se pudo conectar al robot. Verifica que esté encendido.');
            disp(getReport(ME));
        end
    else
        disp('Ya existe una conexión UDP activa.');
    end
end
